presto-docs/src/main/sphinx/optimizer/join-distribution-type.rst

```diff
@@ -5,33 +5,52 @@ Repartitioned and Replicated Joins
-transferred over the network.
+Repartitioned joins are good for larger inputs, as they need less memory on each
+node and allow Presto to handle larger joins overall. However, they can be much
+slower than replicated joins because they can require more data to be transferred
+over the network for tables that differ greatly in size.
 
-Replicated joins can be much faster than repartitioned joins if the replicated table is small enough (especially if it is
-much smaller than the other input). But, if the replicated input is too large, the query can run out of memory.
+Replicated joins can be much faster than repartitioned joins if the replicated
+table is small enough to fit in memory and much smaller than the other input.
+But, if the replicated input is too large, the query can run out of memory.
 
 Choosing the Distribution Type in Presto
 ----------------------------------------
 
-The choice between replicated and repartitioned joins is controlled by the property ``join-distribution-type``. Its possible
-values are ``repartitioned``, ``replicated``, and ``automatic``. The default value is ``automatic``. The property can also be
-set per session using the session property ``join_distribution_type``.
+The choice between replicated and repartitioned joins is controlled by the
+property ``join-distribution-type``. Its possible values are ``repartitioned``,
+``replicated``, and ``automatic``. The default value is ``repartitioned``.
+The property can also be set per session using the session property
+``join_distribution_type``.
 
 .. code-block:: sql
 
-    SET SESSION join_distribution_type = repartitioned;
+    SET SESSION join_distribution_type = 'automatic';
 
-When ``join-distribution-type`` is set to ``repartitioned``, repartitioned distribution is used. When it is set to
-``replicated``, replicated distribution is used. In ``replicated`` mode, the right input to the join is the one that gets
-replicated.
+When ``join-distribution-type`` is set to ``repartitioned``, repartitioned
+distribution is used. When it is set to ``replicated``, replicated distribution
+is used. In ``replicated`` mode, the right input to the join is the one that
+gets replicated.
 
-When the property is set to ``automatic``, the optimizer will choose which type of join to use based on the size of the join
-inputs. If the size of either input to the join is less than 10% of ``query.max-memory-per-node``, Presto will replicate the
-smaller of the inputs. If neither input is small enough or there is insufficient information to determine their size, a
-repartitioned join is performed.
+When the property is set to ``automatic``, the optimizer will choose which type
+of join to use based on the size of the join inputs. The smaller table from a
+join will be replicated if the following two conditions hold:
+
+    1. It is smaller than ``coefficient`` * ``query.max-memory-per-node``
+       The coefficient defaults to  0.01. It can be set using the property
```

Comment:
It might be good to give the rationale behind this 0.01 (10%) default value and how an effective value can be selected based on concurrency.